build-scripts/build_compiler.js

```diff
@@ -0,0 +1,145 @@
+ * that matches the package major version.
+ *
+ * When the COMPILER_NIGHTLY env variable is set, the compiler submodule will be pointing to the latest master
+ * commit. This is for regular integration testing of the compiler with the various tests in this repo's packages.
+ * In this case the compiler will be built as a SNAPSHOT.
+ */
+'use strict';
+
+const {spawn, spawnSync} = require('child_process');
+const ncp = require('ncp');
+const Semver = require('semver');
+
+// Master version number of the packages. Replaces the version
+// information from the root package.json file.
+const packageInfo = require('../lerna.json');
+const packageVersion = new Semver(packageInfo.version);
+
+const fs = require('fs');
+const glob = require('glob');
+
+function getCompilerSubmoduleReleaseVersion() {
+  const gitTagResults = spawnSync('git', ['tag', '--points-at', 'HEAD'], {
+    cwd: './compiler'
+  });
+  if (gitTagResults.status === 0) {
+    const currentTag = gitTagResults.stdout.toString().replace(/\s/g, '');
+    let normalizedTag = currentTag;
+    if (normalizedTag) {
+      // Standard release tags are of the form "vYYYYMMDD".
+      // We also recognize the forms where a special release name is present:
+      //     "webpack-v20180810"
+      //     "v20180810-gwt-fix"
+      normalizedTag = currentTag.replace(/^([a-z]+-)?(v\d{8})(.*)$/, '$2');
+    }
+    return normalizedTag;
+  }
+}
+
+let compilerVersion = getCompilerSubmoduleReleaseVersion();
+if (compilerVersion === `v${packageVersion.major}`) {
+  // Since the tagged commit pom.xml files are still set as SNAPSHOT versions,
+  // update these files to the correct version so that the `compiler --version` command is correct.
+  // TODO(chadkillingsworth): Find a better way to do this
+  const pomPaths = glob.sync('./compiler/**/pom*.xml');
+  pomPaths.forEach(pomPath => {
+    const contents = fs.readFileSync(pomPath, 'utf8');
+    fs.writeFileSync(pomPath, contents.replace('<version>1.0-SNAPSHOT</version>', `<version>${compilerVersion}</version>`), 'utf8');
+  });
+} else {
+  compilerVersion = '1.0-SNAPSHOT';
+}
+
+const compilerJavaBinaryPath = `./compiler/target/closure-compiler-${compilerVersion}.jar`;
+const compilerJsBinaryPath = `./compiler/target/closure-compiler-gwt-${compilerVersion}/jscomp/jscomp.js`;
+
+console.log(process.platform, process.arch, compilerVersion);
+
+function copyCompilerBinaries() {
+  const reportErrors = err => {
+    if (err) {
+      throw new Error(err);
+    }
+  };
+  ncp(compilerJavaBinaryPath, './packages/google-closure-compiler-java/compiler.jar', reportErrors);
+  ncp(compilerJavaBinaryPath, './packages/google-closure-compiler-linux/compiler.jar', reportErrors);
+  ncp(compilerJavaBinaryPath, './packages/google-closure-compiler-osx/compiler.jar', reportErrors);
+  ncp(compilerJsBinaryPath, './packages/google-closure-compiler-js/jscomp.js', reportErrors);
+  ncp('./compiler/contrib', './packages/google-closure-compiler/contrib', reportErrors);
```

Comment:
Although not essential right now, it would be nice if this function were modified to return a Promise that resolves/rejects when the copies are done or fail.
Could you reasonably use an async function, or is that not available in the NodeJS available in Travis?

Reply:
Failing that, perhaps you could make this consistent with the rest of the script and have it do the copies synchronously?

Reply:
Node 6 doesn't support async / await - so for now we can't use it (I so wish we could though ...)